### PR TITLE
[3.5] api: fix duplicate gateway url issue

### PR DESCRIPTION
**Pull request by fuweid**
Run `scripts/genproto.sh --skip-protodoc` and checkout server/wal/walpb/record.pb.go because we still use go.etcd.io/etcd/raft/v3 instead of go.etcd.io/raft/v3.

Before

```
➜  etcd git:(release-3.5) etcdctl put foo bar
OK
➜  etcd git:(release-3.5) etcdctl get -w json foo
{"header":{"cluster_id":14841639068965178418,"member_id":10276657743932975437,"revision":2,"raft_term":2},"kvs":[{"key":"Zm9v","create_revision":2,"mod_revision":2,"version":1,"value":"YmFy"}],"count":1}

➜  etcd git:(release-3.5) etcdctl compact 2
compacted revision 2

➜  etcd git:(release-3.5) http_proxy= curl -L http://127.0.0.1:2379/v3/maintenance/hashkv -X POST -d '{"revsion":2}'
Not Found
```

After

```bash
➜  etcd git:(bp-16464) http_proxy= curl -L http://127.0.0.1:2379/v3/maintenance/hashkv -X POST -d '{"revsion":2}'
{"header":{"cluster_id":"14841639068965178418","member_id":"10276657743932975437","revision":"2","raft_term":"3"},"hash":3305255506,"compact_revision":"2"}%
```

Backport #16464 

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

**Comment by ahrtr** [comment]
Could you also add at least the cases for hash and hashkv from https://github.com/etcd-io/etcd/pull/16489?

**Comment by fuweid** [comment]
> Could you also add at least the cases for hash and hashkv from #16489?

Added. I also increase the timeout for grpcproxy testcase. No idea why it took so long.

File: tests/e2e/v3_curl_test.go [hunk 1]
```diff
@@ -381,6 +381,34 @@ func testV3CurlMaintenanceAlarmMissiongAlarm(cx ctlCtx) {
...
+
+func TestV3CurlMaintenanceHashKV(t *testing.T) {
+	testCtl(t, testV3CurlMaintenanceHashKV, withCfg(*newConfigNoTLS()))
+}
+
+func testV3CurlMaintenanceHashKV(cx ctlCtx) {
+	if err := cURLPost(cx.epc, cURLReq{
+		endpoint: "/v3/maintenance/hashkv",
+		value:    `{"revision": 1}`,
+		expected: `,"revision":"1","raft_term":"2"}`,
```
Comment (ahrtr) [inline tests/e2e/v3_curl_test.go:406]: Unfortunately, 3.5 doesn't support regular expression matching. At least, let's differentiate the response between `/v3/maintenance/hashkv` and `/v3/maintenance/hash`.
```suggestion
		expected: `,"compact_revision":`,
```

**Review by ahrtr — APPROVED** [review]
LGTM

Thanks @fuweid

**Comment by ahrtr** [comment]
Please also add a changelog item for 3.5. thx

**Comment by fuweid** [comment]
> Please also add a changelog item for 3.5. thx

Sure. Will file pr after merged.

**Comment by ahrtr** [comment]
@jmhbnz @serathius  @wenjiaswe

**Review by jmhbnz — APPROVED** [review]
LGTM - Thanks @fuweid

**Comment by fuweid** [comment]
Thanks @ahrtr @jmhbnz for review.

The changelog is here https://github.com/etcd-io/etcd/pull/16503/files

ping @serathius the change looks good to you?

**Pull request merged**